### Plotting NAO data ###

Plotting the NAO for the processed raw and alt_lagged data for the different forecast ranges and different summer and winter NAO.

In [1]:
# Import relevant libraries
import os
import sys
import glob
import re

# Third-party libraries
import numpy as np
import pandas as pd
import xarray as xr
import matplotlib.pyplot as plt

In [2]:
# Import local modules
import functions as func
import nao_skill_functions as nao_func
import nao_alt_lag_functions as nao_lag_func

# Import dictionaries
sys.path.append('/home/users/benhutch/skill-maps')
import dictionaries as dict

# Import the functions from the rose-suite-matching repository
sys.path.append('/home/users/benhutch/skill-maps/rose-suite-matching')
import nao_matching_seasons as nms_func

In [3]:
# Set up the variables
season = "DJFM"
forecast_range = "2-3"
start_year = 1961
end_year = 2014
lag = 4

In [4]:
# Load the raw data for this variable
raw_psl = nao_lag_func.load_data(season=season,
                                  forecast_range=forecast_range,
                                  start_year=start_year,
                                  end_year=end_year,
                                  lag=lag,
                                  alt_lag=False,)

More than one file found
Using the latest datetime file: /gws/nopw/j04/canari/users/benhutch/alternate-lag-processed-data/psl_DJFM_global_1961_2014_2-3_4_1706785429.0699768.npy
Shape of the data: (54, 178, 9, 72, 144)


### Data dimensions ###

In the case of the raw data, the dimensions will be something like:

(54, 178, 9, 72, 144)

* 54 is the number of time steps (e.g. from 1961 to 2014)
* 178 is the number of ensemble members (e.g. from the full CMIP6 ensemble)
* 9 is the number of forecast years (e.g. for s1960, from 1960 to 1970)
* 72 is the number of latitudes (2.5 degree grid spacing)
* 144 is the number of longitudes (2.5 degree grid spacing)

In the case of the alternative lagged (/lagged) data, the dimensions will be something like:

(51, 712, 72, 144)

* 51 is the number of time steps (e.g. from 1964 to 2014)
* 712 is the number of lagged ensemble members (e.g. from the full lagged CMIP6 ensemble)
* 72 is the number of latitudes (2.5 degree grid spacing)
* 144 is the number of longitudes (2.5 degree grid spacing)

In [5]:
# Load the raw data for the alternative lag
hindcast_psl = nao_lag_func.load_data(season=season,
                                      forecast_range=forecast_range,
                                      start_year=start_year,
                                      end_year=end_year,
                                      lag=lag,
                                      alt_lag=False,)

More than one file found
Using the latest datetime file: /gws/nopw/j04/canari/users/benhutch/alternate-lag-processed-data/psl_DJFM_global_1961_2014_2-3_4_1706785429.0699768.npy
Shape of the data: (54, 178, 9, 72, 144)


In [12]:
# import importlib
import importlib

# Import the function again so as not to have to restart the kernel
importlib.reload(sys.modules['nao_alt_lag_functions'])

# import the functions
from nao_alt_lag_functions import *

In [13]:
# Test the other new function
nao_stats = nao_lag_func.calc_nao_stats(data=hindcast_psl,
                                        season=season,
                                        forecast_range=forecast_range,
                                        start_year=start_year,
                                        end_year=end_year,
                                        lag=0,
                                        alt_lag=False,
                                        )

Using standard NAO definition


/apps/jasmin/jaspy/miniconda_envs/jaspy3.10/m3-4.9.2/envs/jaspy3.10-m3-4.9.2-r20220721/lib/python3.10/site-packages/iris/coords.py:2220: UserWarning: Collapsing a non-contiguous coordinate. Metadata may not be fully descriptive for 'time'.
  warnings.warn(msg.format(self.name()))


Time dimension of obs: ['1960-12-31T00:00:00.000000000' '1961-12-31T00:00:00.000000000'
 '1962-12-31T00:00:00.000000000' '1963-12-31T00:00:00.000000000'
 '1964-12-31T00:00:00.000000000' '1965-12-31T00:00:00.000000000'
 '1966-12-31T00:00:00.000000000' '1967-12-31T00:00:00.000000000'
 '1968-12-31T00:00:00.000000000' '1969-12-31T00:00:00.000000000'
 '1970-12-31T00:00:00.000000000' '1971-12-31T00:00:00.000000000'
 '1972-12-31T00:00:00.000000000' '1973-12-31T00:00:00.000000000'
 '1974-12-31T00:00:00.000000000' '1975-12-31T00:00:00.000000000'
 '1976-12-31T00:00:00.000000000' '1977-12-31T00:00:00.000000000'
 '1978-12-31T00:00:00.000000000' '1979-12-31T00:00:00.000000000'
 '1980-12-31T00:00:00.000000000' '1981-12-31T00:00:00.000000000'
 '1982-12-31T00:00:00.000000000' '1983-12-31T00:00:00.000000000'
 '1984-12-31T00:00:00.000000000' '1985-12-31T00:00:00.000000000'
 '1986-12-31T00:00:00.000000000' '1987-12-31T00:00:00.000000000'
 '1988-12-31T00:00:00.000000000' '1989-12-31T00:00:00.000000000'
 '

In [14]:
print(nao_stats)

{'obs_nao': array([-1256.1571  , -1080.2263  , -1164.7241  ,  -472.65207 ,
           3.959549, -1180.1301  , -1277.0947  ,  -512.74207 ,
        -102.48291 ,   513.81665 ,   565.6443  ,   270.06604 ,
         407.28934 ,  -293.86163 ,  -809.7915  ,  -711.35913 ,
        -594.40283 ,    14.419922,    43.514606,   419.3222  ,
         725.20825 ,   -62.96    ,  -389.37643 ,  -307.0416  ,
        -399.3613  ,   623.6166  ,  1282.9071  ,   686.5044  ,
         560.6515  ,   839.6625  ,   931.1673  ,  1041.9651  ,
          37.9777  ,  -627.19476 ,  -152.48755 ,   280.73767 ,
         759.2278  ,  -102.454346,  -600.73193 ,  -136.73303 ,
        -219.1565  ,  -124.405106,  -199.41695 ,   152.41072 ,
         694.4708  ,   383.4062  ,  -961.8202  , -1322.834   ,
         341.26678 ,    73.80237 ,   -99.327065,  1372.7769  ,
        1202.7002  ,   520.7555  ], dtype=float32), 'model_nao_mean': array([  21.87170895,  -46.36442752,  -80.11203476, -154.55325777,
         33.40027944,  -57.84025